stdlib/gzip.pyi

```diff
@@ -159,9 +159,9 @@ class _GzipReader(_compression.DecompressReader):
     def __init__(self, fp: _ReadableFileobj) -> None: ...
 
 if sys.version_info >= (3, 8):
-    def compress(data: bytes, compresslevel: int = ..., *, mtime: float | None = ...) -> bytes: ...
+    def compress(data: ReadableBuffer, compresslevel: int = ..., *, mtime: float | None = ...) -> bytes: ...
```

Comment:
This also does `len(data)` in one branch, so we should use `_BufferWithLen`.

Reply:
Done!